src/pages/Login.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState, ChangeEvent } from "react";
+import styled from "@emotion/styled/macro";
+import { useAuthDispatchContext, useAuthStateContext } from "../context/auth";
+import { Redirect, RouteComponentProps } from "react-router-dom";
+import { Location } from "history";
+
+const getReferer = (location: Location) => {
+  if (location && location.state && "referer" in location.state) {
```

Comment:
You can refactor the body of this function to: `return location.state.referer ?? "/";`

Reply:
i'm afraid that does not work because the state type in react router is an empty interface. therefore this annoying way of deriving its value must be done.

Reply:
You could do something like this: 
```
const getReferer = (location: Location<{ refer?: String }>) => {
  return location.state.refer ?? "/";
};
```